admin_guide/node_problem_detector.adoc

```diff
@@ -0,0 +1,511 @@
+----
+
+To use the node problem detector, use the `oc get` command with a resource:
+
+[source,bash]
+----
+oc get node registry-router-1 -o yaml | grep -B5 KernelDeadlock
+
+ - lastHeartbeatTime: 2018-04-17T08:01:44Z
+    lastTransitionTime: 2018-04-17T08:01:43Z
+    message: task docker:1234 blocked for more than 700 seconds.
+    reason: DockerHung
+    status: "True"
+----
+
+[[admin-guide-node-problem-detector-run]]
+== Reinstalling the Node Problem Detector
+
+If for any reason, you need to re-create the xref:../dev_guide/daemonsets#dev-guide-daemonsets[daemonset] object, run the following Ansible playbook:
+
+[source,bash]
+----
+ansible-playbook ~/openshift-ansible/playbooks/openshift-node-problem-detector/config.yml
+----
+
+You can also manually create the daemonset object, using the following steps. For {product-title} 
+Google Compute Engine (GCE) infrastructures, see xref:[Running the Node Problem Detector in a GCE Cluster].
+
+When xref:../dev_guide/daemonsets.adoc#dev-guide-daemonsets[creating daemonsets], the `nodeSelector` field indicates the nodes on which the daemonset should deploy.
+
+. Create a Daemonset object for the node problem detector on your master:
++
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: node-problem-detector
+spec:
+  selector:
+    matchLabels:
+      name: node-problem-detector
+  template:
+    metadata:
+      labels:
+        app: node-problem-detector
+    spec:
+      containers:
+      - name: node-problem-detector
+        command:
+        - node-problem-detector
+        - --address=127.0.0.1 <1>
+        - --port=20256 <1>
+        - --logtostderr
+        - --kernel-monitor=/config/kernel-monitor.json <2>
+        image: k8s.gcr.io/node-problem-detector:v0.2
+        imagePullPolicy: Always
+        securityContext:
+          privileged: true
+        env:
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+        volumeMounts:
+        - name: log
+          mountPath: /var/log
+          readOnly: true
+        - name: kmsg
+          mountPath: /dev/kmsg
+          readOnly: true
+        # Make sure node problem detector is in the same timezone
+        # with the host.
+        - name: localtime
+          mountPath: /etc/localtime
+          readOnly: true
+        - name: config
+          mountPath: /config
+          readOnly: true
+      volumes:
+      - name: log
+        # Config `log` to your system log directory
+        hostPath:
+          path: /var/log/ <3>
+      - name: kmsg
+        hostPath:
+          path: /dev/kmsg
+      - name: localtime
+        hostPath:
+          path: /etc/localtime
+      - name: config
+        configMap:
+          name: node-problem-detector-config <4>
+----
```

Comment:
I'm not sure what to do here. If we want to provide a yaml file, it should probably be very close to the ones we ship for Origin and OCP.  Here's the one for Origin, but do we really want to have to keep this up to date with the right image spec?

```
apiVersion: v1
items:
- apiVersion: apps/v1
  kind: DaemonSet
  metadata:
    labels:
      app: node-problem-detector
    name: node-problem-detector
  spec:
    template:
      metadata:
        labels:                                                                                                                                                                                                                             
          name: node-problem-detector                                                                                                                                                                                                       
      spec:                                                                                                                                                                                                                                 
        containers:                                                                                                                                                                                                                         
        - command:                                                                                                                                                                                                                          
          - node-problem-detector                                                                                                                                                                                                           
          - --logtostderr                                                                                                                                                                                                                   
          - --system-log-monitors=/etc/npd/kernel-monitor.json,/etc/npd/docker-monitor.json                                                                                                                                                 
          env:                                                                                                                                                                                                                              
          - name: NODE_NAME                                                                                                                                                                                                                 
            valueFrom:                                                                                                                                                                                                                      
              fieldRef:                                                                                                                                                                                                                     
                apiVersion: v1                                                                                                                                                                                                              
                fieldPath: spec.nodeName                                                                                                                                                                                                    
          image: docker.io/openshift/node-problem-detector:v3.10.0                                                                                                                                                                           
          imagePullPolicy: IfNotPresent                                                                                                                                                                                                     
          name: node-problem-detector                                                                                                                                                                                                       
          resources: {}                                                                                                                                                                                                                     
          securityContext:                                                                                                                                                                                                                  
            privileged: true                                                                                                                                                                                                                
          terminationMessagePath: /dev/termination-log                                                                                                                                                                                      
          terminationMessagePolicy: File                                                                                                                                                                                                    
          volumeMounts:                                                                                                                                                                                                                     
          - mountPath: /host/log                                                                                                                                                                                                            
            name: log                                                                                                                                                                                                                       
            readOnly: true                                                                                                                                                                                                                  
          - mountPath: /etc/localtime
            name: localtime
            readOnly: true
          - mountPath: /etc/npd
            name: config
        restartPolicy: Always
        securityContext: {}
        serviceAccount: node-problem-detector
        serviceAccountName: node-problem-detector
        terminationGracePeriodSeconds: 30
        volumes:
        - hostPath:
            path: /var/log/
            type: ""
          name: log
        - hostPath:
            path: /etc/localtime
            type: ""
          name: localtime
        - configMap:
            name: node-problem-detector
          name: config
    updateStrategy:
      type: RollingUpdate
```